### PR TITLE
[incubator-kie-issues#1409]-Fixed matches function behavior

### LINKED ISSUE #1409: `matches()` function wrongly behaves - checks and unicode
TCK Tests revealed some cases where `matches()` wrongly behaves.

1.  `matches("abracadabra", "bra", "p")` should return an error, as the "p" flag doesn't exist; (Please check all the functions where the flags parameter is used e.g. `replace()`)
2. `matches("\u212A", "k", "i")` should return true. \u212A is the [Kelvin Sign](https://codepoints.net/U+212A?lang=en), basically a K. Before fixing that, double-check in the specs that this is the expected behavior 
3. `matches("input", "pattern", " ")` same as point 1.
4. `matches("input", "pattern", "X")` same as point 1.
5.  `matches("hello world", "hello\ sworld", "x")`The `x` is not supported. (Please check all the functions where the flags parameter is used e.g. `replace()`)
6. `matches("h", "(.)\3")` `matches("h", "(.)\2")` and `matches("input", "\3")` should fail. Please investigate why they should fail

### PR DESCRIPTION
**Pull request by bncriju**
Fixes https://github.com/apache/incubator-kie-issues/issues/1409

<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request and downstream checks</b>  
  - Push a new commit to the PR. An empty commit would be enough.

- for a <b>full downstream build</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- for <b>Jenkins PR check only</b>
  - If you are an ASF committer for KIE podling, login to Jenkins (https://ci-builds.apache.org/job/KIE/job/drools/), go to the specific PR job, and click on `Build Now` button.
</details>

### CODE REVIEW

**Review by gitgabrio — APPROVED** [review]
Many thanks @bncriju great job!!!

**Review by baldimir — APPROVED** [review]

File: kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/functions/MatchesFunction.java [hunk 1]
```diff
@@ -18,16 +18,21 @@
  */
 package org.kie.dmn.feel.runtime.functions;
 
+import java.security.InvalidParameterException;
```
Comment (yesamer) [inline kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/functions/MatchesFunction.java:21]: @bncriju is this import used / correct?

Comment (bncriju) [reply kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/functions/MatchesFunction.java:21]: @yesamer Whenever input/pattern is null, this is the most appropriate one to be used. it seems. This is used to catch this condition and whenever we verify this case in test as well.

Comment (yesamer) [reply kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/functions/MatchesFunction.java:21]: @bncriju My point is: that expection is part of the `java.security` package. That means it should be used in the security management scope (as described here https://docs.oracle.com/javase/8/docs/api/java/security/InvalidParameterException.html). 
In our context, [IllegalArgumentException](https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalArgumentException.html) should be used

**Comment by yesamer** [comment]
@bncriju  I run your changes against the TCK tests, and cases 5. and 6. are still not working. Is that intended?

**Pull request merged**